dmriprep/interfaces/vectors.py

```diff
@@ -1,91 +1,101 @@
+    """
+    Ensure the correctness of the gradient table.
+
+    Example
+    -------
+
+    >>> os.chdir(tmpdir)
+    >>> check = CheckGradientTable(
+    ...     dwi_file=str(data_dir / 'dwi.nii.gz'),
+    ...     in_rasb=str(data_dir / 'dwi.tsv')).run()
+    >>> check.outputs.pole
+    (0.0, 0.0, 0.0)
+    >>> check.outputs.full_sphere
+    True
+
+    >>> check = CheckGradientTable(
+    ...     dwi_file=str(data_dir / 'dwi.nii.gz'),
+    ...     in_bvec=str(data_dir / 'bvec'),
+    ...     in_bval=str(data_dir / 'bval')).run()
+    >>> check.outputs.pole
+    (0.0, 0.0, 0.0)
+    >>> check.outputs.full_sphere
+    True
+    >>> newrasb = np.loadtxt(check.outputs.out_rasb, skiprows=1)
+    >>> oldrasb = np.loadtxt(str(data_dir / 'dwi.tsv'), skiprows=1)
+    >>> np.allclose(newrasb, oldrasb, rtol=1.e-3)
+    True
+
+    """
+
+    input_spec = _CheckGradientTableInputSpec
+    output_spec = _CheckGradientTableOutputSpec
 
     def _run_interface(self, runtime):
-        from dmriprep.utils.vectors import VectorTools
-        if isdefined(self.inputs.fbval) and isdefined(self.inputs.fbvec):
-            fbval = self.inputs.fbval
-            fbvec = self.inputs.fbvec
-        else:
-            fbval = None
-            fbvec = None
-
-        if isdefined(self.inputs.rasb_tsv_file):
-            rasb_tsv_file = self.inputs.rasb_tsv_file
-        else:
-            rasb_tsv_file = None
-
-        if isdefined(self.inputs.basedir):
-            basedir = self.inputs.basedir
-        else:
-            basedir = runtime.cwd
-
-        vt = VectorTools(basedir, self.inputs.dwi_file, fbval, fbvec, rasb_tsv_file, self.inputs.B0_NORM_EPSILON,
-                         self.inputs.bval_scaling)
-
-        # Read in vectors
-        if rasb_tsv_file is not None:
-            vt.read_rasb_tsv()
-        elif fbval is not None and fbvec is not None:
-            vt.read_bvals_bvecs()
-        else:
-            raise OSError('VectorTools methods require either fbval/fbvec input files or an input rasb .tsv file.')
-
-        # Rescale vectors
-        if self.inputs.rescale is True:
-            vt.rescale_vecs()
-
-        # Build gradient table
-        gtab, b0_ixs = vt.build_gradient_table()
-
-        # Check vector integrity
-        is_hemi, pole, slm = vt.checkvecs()
-
-        if self.inputs.save_fsl_style is True:
-            fbval_out_file, fbvec_out_file = vt.save_vecs_fsl()
-        else:
-            fbval_out_file = None
-            fbvec_out_file = None
-
-        self._results['gtab'] = gtab
-        self._results['is_hemi'] = is_hemi
-        self._results['pole'] = pole
-        self._results['slm'] = slm
-        self._results['b0_ixs'] = b0_ixs
-        self._results['fbval_out_file'] = fbval_out_file
-        self._results['fbvec_out_file'] = fbvec_out_file
-        self._results['rasb_tsv_out_file'] = vt.write_rasb_tsv()
+        rasb_file = _undefined(self.inputs, 'in_rasb')
+
+        table = DiffusionGradientTable(
```

Comment:
this is why you want the right data structure in place: it makes it look so easy.